src/main/java/com/twilio/http/Request.java

```diff
@@ -174,9 +174,14 @@ private String buildURL() {
 
                 host = joinIgnoreNull(".", product, targetEdge, targetRegion, domain);
             }
-
-            return new URI(parsedUrl.getProtocol(), parsedUrl.getUserInfo(), host, parsedUrl.getPort(),
-                    parsedUrl.getPath(), parsedUrl.getQuery(), parsedUrl.getRef()).toURL().toString();
+            String getQuery = "";
+            if( parsedUrl.getQuery() != null ) {
+                getQuery = "?" + parsedUrl.getQuery();
+            }
+            String constructedURI = new URI(
+                    parsedUrl.getProtocol(), parsedUrl.getUserInfo(), host, parsedUrl.getPort(),
+                    parsedUrl.getPath(), null, parsedUrl.getRef()).toString() + getQuery;
```

Comment:
https://en.wikipedia.org/wiki/URI_fragment
> The fragment identifier introduced by a hash mark `#` is the optional last part of a URL for a document.

This places it _before_ the query.

What about just decoding the query part before constructing the URI?